### DISCUSSION TITLE
How to get Windows 7 style title bar buttons?

### DISCUSSION THREAD

**Discussion by MD7070B** [ALTaleX531/OpenGlass#67] [Q&A]
Installed OpenGlass which has successfully added a nice transparent effect to titlebars & window borders, on Windows 10 22H2. What I would really like is to be able to restore Windows 7 titlebar buttons, and also the rounded window borders (although less important). 

I can't seem to find any documentation anywhere showing exactly how to get Windows 7 titlebar buttons. I have tried pointing `CustomThemeMsstyle` to an `aero.msstyle` file, extracted from a Windows 7 ISO, but this did absolutely nothing. Would be seriously glad of any help. 

Also, regarding the rounded borders, I have played around with the registry, and have managed to round the borders slightly, however I have noticed that the windows are still actually square - it seems to be the shading within that is rounded - thus there is still an annoying square-cornered outline present. Any way of removing this?

**Comment by AshtyBoi** [comment]
You can use Aero Window Manager for the Windows 7 titlebar buttons and the rounded corners.

**Reply by MD7070B** [reply]
So you mean people with screenshots of glass effect titlebars AND Windows 7 style buttons are using both OpenGlass & AWM? Any guides for setting up AWM, when I just tried it, it shows `ERROR 0` a couple of times in the console, and does not do anything useful. (Changes titlebars slightly but certainly does not look great).

**Reply by DAPOPOBEFASTONYOAZZ** [reply]
If you have seen any of my screenshots on other discussions for this project, I have been using AWM for window buttons. If you would, please remind me to revisit this at a later time when I have more time and I will detail an installation of AWM for you and any other users that may have questions.

**Reply by Tamim468** [reply]
You don't have to use AWM for Windows 7 titlebar buttons anymore, Openglass recently got a pull request merged which means it now has proper caption buttons feature and others like inactive opacite and icon margin, check it out!

Note: the latest release currently doesn't have these features, so you will have to compile the source code manually.
![image](https://github.com/user-attachments/assets/8c3a16d1-b4e1-4608-851c-72757b1b666b)

**Reply by MD7070B** [reply]
Looks good! I don't have time to compile to binaries myself right now, so any idea how long until this is likely to make it into a release?

**Reply by DAPOPOBEFASTONYOAZZ** [reply]
> Looks good! I don't have time to compile to binaries myself right now, so any idea how long until this is likely to make it into a release?

This actually is in the latest release of 2.1.0. Tamim was talking about 2.0.5, 2.1.0 wasn't out just yet.

**Reply by DAPOPOBEFASTONYOAZZ** [reply]
> You don't have to use AWM for Windows 7 titlebar buttons anymore, Openglass recently got a pull request merged which means it now has proper caption buttons feature and others like inactive opacite and icon margin, check it out!

The only reason I use AWM still is because the latest build only uses the Windows 7 button sizes, but I want the Windows Vista sizes. Vista's buttons are slightly smaller. AWM also provides controls for inactive font color and more authentic titlebar text glow.

**Reply by MD7070B** [reply]
Ideal - thanks! I just removed the old version I was running and installed the latest release - 2.1.0. 

However, (and please excuse my ignorance!), I still have the default titlebar buttons, the the glass effect / transparency seems to be going a bit wrong, especially on active Windows - the transparency was there for a few seconds, but then this occured:

![Capture](https://github.com/user-attachments/assets/f87ccb08-15b6-4f9d-afaf-8b71df283d8a)

Any idea what I am doing wrong?
Thanks for your help!

**Reply by DAPOPOBEFASTONYOAZZ** [reply]
> Ideal - thanks! I just removed the old version I was running and installed the latest release - 2.1.0.
> 
> However, (and please excuse my ignorance!), I still have the default titlebar buttons, the the glass effect / transparency seems to be going a bit wrong, especially on active Windows - the transparency was there for a few seconds, but then this occured:
> 
> Any idea what I am doing wrong? Thanks for your help!

Hard to say unless we could take a look at your config. Could you please share a screenshot of your DWM config? You can find it by opening Registry Editor and going to `HKEY_CURRENT_USER\SOFTWARE\Microsoft\Windows\DWM` and once there, the pane to your right will show all of your registry keys related to how you have OpenGlass configured. Please also be sure to blur or remove any personal information as well!

**Reply by MD7070B** [reply]
Certainly! Here is the config in `Computer\HKEY_CURRENT_USER\SOFTWARE\Microsoft\Windows\DWM`. Its a long time since I did anything with these values, probably back before Glass8 broke / disappeared - I remember I eventually got Glass8 working well, possibly by loading an atlas file, but my memory fails me...

![DWM_Reg_Keys](https://github.com/user-attachments/assets/314538cd-d36c-4e29-958f-6dae90f294d1)

**Reply by DAPOPOBEFASTONYOAZZ** [reply]
> Certainly! Here is the config in `Computer\HKEY_CURRENT_USER\SOFTWARE\Microsoft\Windows\DWM`. Its a long time since I did anything with these values, probably back before Glass8 broke / disappeared - I remember I eventually got Glass8 working well, possibly by loading an atlas file, but my memory fails me...

I see the issue, looks like you are trying to use the "Sky" color. Recently, OpenGlass added `ColorizationBlurBalanceOverride` in 2.0 I believe, and you don't have this set yet. Your registry keys should be:

`ColorizationAfterglowBalanceOverride` = 0x0000002b (43)
`ColorizationBlurBalanceOverride` = 0x00000031 (49)
`ColorizationColorBalanceOverride` = 0x0000008 (8)

`GlassOpacity` was used in the master branch instead of `ColorizationBlurBalanceOverride`. `GlassOpactiy` and `GlassOpacityInactive` only apply to `GlassType` 0 (Windows Vista) now, and not 1 (Windows 7.)

Don't forget to set your `ColorizationAfterglow`, `ColorizationColor`, and `ColorizationColorInactive` registry keys so the color shows properly too. If you don't, it will still look off. The "Sky" color for these values is `0xc474b8fc`.

And OpenGlass does support theme atlases as well, but my choice is to use a custom theme by patching my UXstyles for familiarity and ease reasons. If you know how to import a theme atlas, you can still do that.

You can also always check the home page of the project for a table with all of the supported registry keys, value settings and an explanation for what they do.

Happy modding!

**Reply by MD7070B** [reply]
Thanks for your help - it has certainly improved the result! After applying the above registry changes, I have this:

![image](https://github.com/user-attachments/assets/7c47ed76-73ce-457d-88a5-9bd1b48b7749)

Does that color look OK to you? Strangely, I thought it was a little lighter blue before I tried tweaking the reflection (below), but I think I must have imagined that, as it didn't revert when I disabled the reflection setting...

I found the `ColorizationGlassReflectionIntensity` and `ColorizationGlassReflectionIntensityInactive` options on the project home page, and enabled this, which gave a subtle 'striped' reflection effect, which I like:

![image](https://github.com/user-attachments/assets/e0b6ec09-1722-42fa-ba53-d7fe3544c26d)

Finally, I did try a `CustomThemeAtlas` in an attempt to restore Windows 7 borders & title bar buttons. - I was having trouble finding one that worked, as most of them caused the title bar to render completely wrong. 
This one (https://drive.google.com/drive/folders/1luiRyy6Q6ZFhMvruCscpl88TgMxEep6y) is the best I have found so far (actually the only one which works), but it still isn't quite perfect:

![image](https://github.com/user-attachments/assets/d7e72763-9416-4447-b7e3-f841401ed183)

**EDIT:**
I have just noticed one other issue with the theme atlas mentioned above: the title bar buttons don't really render properly when windows are maximized - there height seems a bit off...
![image](https://github.com/user-attachments/assets/431b187f-85c8-44b4-a83d-e9d1009d4527)

**Reply by LittleFox2024** [reply]
That's a normal issue with Windows 10 and 11. They changed it so that the title bar is bigger when it is not maximized (probably for touch reasons). OpenGlass just added an option to restore title bar size. Create a DWORD called "CaptionButtons" and set the value to 2 (for Windows 7 style).

https://github.com/ALTaleX531/OpenGlass/?tab=readme-ov-file#theme-settings

**Reply by MD7070B** [reply]
Thanks!

However, strangely, using any value other than `0x0`, which is Windows 10 Default, actually makes the problem worse. This problem now occurs on maximized *and* not maximized windows:

![image](https://github.com/user-attachments/assets/07da0420-09cf-478d-a088-9839fadf255c)

Suppose it's consistent at least, but not really what I am going for...

Could it be an issue with the theme atlas which I am using, being incompatible with the `CaptionButtons` option? This is my current config:
![image](https://github.com/user-attachments/assets/d5f6a9b9-38fa-4eb4-88f2-5cfbd2332cc4)

**Reply by DAPOPOBEFASTONYOAZZ** [reply]
> Thanks for your help - it has certainly improved the result! After applying the above registry changes, I have this:
>
> Does that color look OK to you? Strangely, I thought it was a little lighter blue before I tried tweaking the reflection (below), but I think I must have imagined that, as it didn't revert when I disabled the reflection setting...
> 
> I found the `ColorizationGlassReflectionIntensity` and `ColorizationGlassReflectionIntensityInactive` options on the project home page, and enabled this, which gave a subtle 'striped' reflection effect, which I like:
> 
> Finally, I did try a `CustomThemeAtlas` in an attempt to restore Windows 7 borders & title bar buttons. - I was having trouble finding one that worked, as most of them caused the title bar to render completely wrong. This one (https://drive.google.com/drive/folders/1luiRyy6Q6ZFhMvruCscpl88TgMxEep6y) is the best I have found so far (actually the only one which works), but it still isn't quite perfect:
> 
> **EDIT:** I have just noticed one other issue with the theme atlas mentioned above: the title bar buttons don't really render properly when windows are maximized - there height seems a bit off... ![image](https://private-user-

When I have a chance, I will share more of my config with you to help you get set up properly.

**Reply by MD7070B** [reply]
Much appreciated!

**Reply by ALTaleX531** [reply]
@MD7070B Your theme atlas is terrible, change it. I can share the one extracted from aero10 if @vaporvance allows it.

**Reply by MD7070B** [reply]
> @MD7070B Your theme atlas is terrible, change it. I can share the one extracted from aero10 if @vaporvance allows it.

I suspected as much. More than willing to change it, and would be great if you could share a good one - my current one is just the first I found which actually rendered semi-properly, and then I ran out of time to search for a better one...

**Reply by LittleFox2024** [reply]
That looks like it has smaller title bar buttons built in. I even did that in one version of my theme in an attempt to get the size correct. The issue is that what I believe restoring Windows 7 sizes does is it shrinks the buttons by a specific amount. However, it's built for the default Windows 10/11 size. Which means that if the buttons are already small normally, they're going to look miniscule with the change.

It's why in my theme I had two version - small title bar buttons and normal title bar. The normal title bar looked weird normally but it worked well with these fixes, while small title bar buttons were better in normal usage without a mod (although they still looked tiny when maximized... nothing I could do about it in a visual style).

**Comment by MD7070B** [comment]
This is still working quite nicely - albeit with various issues here and there. If anyone does have or know of a good Windows 7 Aero Glass theme atlas file, I would be glad to hear of it!

**Reply by DAPOPOBEFASTONYOAZZ** [reply]
> This is still working quite nicely - albeit with various issues here and there. If anyone does have or know of a good Windows 7 Aero Glass theme atlas file, I would be glad to hear of it!

Sorry I haven't gotten back to you sooner on this, I've been rather busy.

If you're on Windows 10, The best thing I could recommend is downloading the Aero10 theme, using Resource Hacker to open up the theme file > going into the stream folder > right click on the atlas and "Save as PNG." This will get you a good theme atlas. I won't realize my custom one without @vaporvance's permission. It's a blend of Vista and 7 without the corner highlights because they're broken in the current state so hopefully they'll be implemented as a functionality of OpenGlass (see #152.) However it does retain the side border shine.

Although, personally I would say just go ahead and patch your uxtheme.dll using UltraUXThemePatcher and just install the themes and run it that way. You'll save yourself a headache of configuration (in my personal opinion.)

**Comment by MD7070B** [comment]
No problem about the delay - I have also been very busy. 

I will explore both of your options - will have a look for an available copy of Aero10.

P.S: Which is the 'official' download for UXtheme? [https://mhoefs.eu/software_uxtheme.php](https://mhoefs.eu/software_uxtheme.php?ref=syssel&lang=en), [https://www.ultrauxthemepatcher.com/](https://www.ultrauxthemepatcher.com/), both or neither?

**Reply by DAPOPOBEFASTONYOAZZ** [reply]
> No problem about the delay - I have also been very busy.
> 
> I will explore both of your options - will have a look for an available copy of Aero10.
> 
> P.S: Which is the 'official' download for UXtheme? [https://mhoefs.eu/software_uxtheme.php](https://mhoefs.eu/software_uxtheme.php?ref=syssel&lang=en), https://www.ultrauxthemepatcher.com/, both or neither?

You can download Aero10 here: https://www.deviantart.com/vaporvance/art/Aero10-for-Windows-10-1903-22H2-909711949

The second link you provided is the official download for UltraUXThemePatcher.

**Reply by MD7070B** [reply]
Thanks for the information. I will try this at some point, am currently unable to access [https://www.deviantart.com/](https://www.deviantart.com/) though, think I must be having some network / proxy issues.
